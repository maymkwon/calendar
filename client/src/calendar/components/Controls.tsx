import React, { Component } from 'react';
import { MdChevronLeft, MdChevronRight } from 'react-icons/md';
import { SCArrowBtn, SCBasicBtn } from '../../style/Buttons';
import styled from 'styled-components';
import moment, { Moment as MomentType } from 'moment';
import { componentView } from '../../utils/constants';
import * as dateUtils from '../../utils/dates';
import cn from 'classnames';

const SCWrap = styled.div.attrs({ className: 'c-control dfacjsb' })`
  padding: 0 20px;
  border-bottom: 1px solid #eee;
  position: sticky;
  top: 0;
  background: #fff;
  z-index: 999;
  .c-control {
    &__term {
      .basic-btn:first-child {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
      }
      .basic-btn:last-child {
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
      }
      .basic-btn + .basic-btn {
        border-left: 0;
      }
    }

    &__currnet {
      font-size: 24px;
      margin-left: 20px;
    }
  }
`;

interface Props {
  date: MomentType;
  view: string;
  onChangeDate: (string: string) => void;
  onSelectDate: (MomentType: MomentType) => void;
  onChangeView: (string: string) => void;
}

class Controls extends Component<Props> {
  render() {
    const { view, date, onChangeDate, onSelectDate, onChangeView } = this.props;
    const weekOfMonth = dateUtils.getWeekOfMonth(date);
    return (
      <SCWrap id="controls">
        <div className="dfac">
          <SCArrowBtn
            className="dfacjcc"
            size={50}
            onClick={() => onChangeDate('prev')}
          >
            <MdChevronLeft size={30} />
          </SCArrowBtn>
          <SCArrowBtn
            className="dfacjcc"
            size={50}
            onClick={() => onChangeDate('next')}
          >
            <MdChevronRight size={30} />
          </SCArrowBtn>
          <div
            className="c-control__currnet"
            role="button"
            onClick={() => onSelectDate(moment())}
          >
            {date.format('YYYY 년 MM 월')}
            {view === componentView.WEEK && (
              <span style={{ marginLeft: 10 }}>{weekOfMonth}주</span>
            )}
          </div>
        </div>

        <div className="c-control__term">
          <SCBasicBtn
            className={cn({ active: view === componentView.MONTH })}
            onClick={() => onChangeView(componentView.MONTH)}
          >
            월
          </SCBasicBtn>
          <SCBasicBtn
            className={cn({ active: view === componentView.WEEK })}
            onClick={() => onChangeView(componentView.WEEK)}
          >
            주
          </SCBasicBtn>
        </div>
      </SCWrap>
    );
  }
}

export default Controls;
