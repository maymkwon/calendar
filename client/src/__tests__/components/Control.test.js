import React from 'react';
import { mount } from 'enzyme';
import { shallow } from 'enzyme';

import Controls from '../../calendar/components/Controls';
import moment from 'moment';

describe('<Controls /> 컴포넌트', () => {
  it('마운트 컴포넌트', () => {
    let date = moment('2019-10-16');
    let view = 'month';

    const wrapper = mount(<Controls date={date} view={view} />);
    expect(wrapper.props().date).toBe(date);
    expect(wrapper.props().view).toBe(view);
  });
  it('렌더 컴포넌트', () => {
    let date = moment('2019-10-16');

    const wrapper = shallow(<Controls date={date} />);
    const render = wrapper.render();

    expect(render.find('.c-control__currnet').text()).toBe(
      date.format('YYYY 년 MM 월')
    );
  });
});
