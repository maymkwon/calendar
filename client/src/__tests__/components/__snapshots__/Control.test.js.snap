// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> render() renders the component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        Object {
          "className": "c-control dfacjsb",
        },
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  padding: 0 20px;
  border-bottom: 1px solid #eee;
  position: sticky;
  top: 0;
  background: #fff;
  z-index: 999;
  .c-control {
    &__term {
      .basic-btn:first-child {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
      }
      .basic-btn:last-child {
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
      }
      .basic-btn + .basic-btn {
        border-left: 0;
      }
    }

    &__currnet {
      font-size: 24px;
      margin-left: 20px;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  id="controls"
>
  <div
    className="dfac"
  >
    <Styled(styled.button)
      className="dfacjcc"
      onClick={[Function]}
      size={50}
    >
      <MdChevronLeft
        size={30}
      />
    </Styled(styled.button)>
    <Styled(styled.button)
      className="dfacjcc"
      onClick={[Function]}
      size={50}
    >
      <MdChevronRight
        size={30}
      />
    </Styled(styled.button)>
    <div
      className="c-control__currnet"
      onClick={[Function]}
      role="button"
    >
      2019 년 10 월
    </div>
  </div>
  <div
    className="c-control__term"
  >
    <Styled(styled.button)
      className=""
      onClick={[Function]}
    >
      월
    </Styled(styled.button)>
    <Styled(styled.button)
      className=""
      onClick={[Function]}
    >
      주
    </Styled(styled.button)>
  </div>
</StyledComponent>
`;
